### PR TITLE
Data generation

### PR DESCRIPTION
**Pull request by clfhhc**

### CODE REVIEW

File: mock-data/generateData.js [hunk 1]
```diff
@@ -1,120 +1,155 @@
...
-    const numOfListingReviews = chance.integer({ min: 1, max: maxReviews });
-    console.log(numOfListingReviews);
-    const currentListingID = i;
-    for (let j = 0; j < numOfListingReviews; j += 1) {
-      allListingReviews.push(createReview(currentListingID));
+const generateAllReviews =  (listingStart, listingEnd, callback) => {
+  for (let i = listingStart; i <= listingEnd; i++) {
+    const numOfListingReviews = generateInteger(MAX_REVIEWS_DEFAULT, 1);
+    for (let j = 0; j < numOfListingReviews; j++) {
+      if (callback)  callback(createReview(i));
```
Comment (briansai) [inline mock-data/generateData.js:94]: curly brackets around callback(createReview(i)) in if statements for best practice.

File: mock-data/generateData.js [hunk 2]
```diff
@@ -1,120 +1,155 @@
...
-    const maxGuests = chance.integer({ min: 1, max: 10 });
+    const listingName = chance.word();
+    const maxGuests = generateInteger(10, 1);
     const price = costModifier(35, 150, maxGuests);
-    const minStay = chance.integer({ min: MIN_STAY_DEFAULT, max: MAX_STAY_DEFAULT });
+    const minStay = generateInteger(MAX_STAY_DEFAULT, MIN_STAY_DEFAULT);
     const cleaningFee = costModifier(5, 15, maxGuests);
     const areaTax = costModifier(5, 15, maxGuests);
-    allListingInfo.push([owner, maxGuests, price, minStay, cleaningFee, areaTax]);
+    if (callback) callback(`${owner},${listingName},${maxGuests},${price},${minStay},${cleaningFee},${areaTax}\n`);
```
Comment (briansai) [inline mock-data/generateData.js:121]: Same as above.  Curly brackets for if statements for best practice.

File: mock-data/generateData.js [hunk 3]
```diff
@@ -1,120 +1,155 @@
...
-    user.split('').forEach((char) => {
-      double += char;
-    });
-    accounts.push(double);
+const generateOneBooking = (listingId) => addMockBookings(listingId, START_DATE, 90);
+
+const generateFirstName =  (listingStart, listingEnd, callback) => {
+  const userRatio = USERS_DEFAULT / LISTINGS_DEFAULT;
+  for (let i = (listingStart - 1) * userRatio + 1 ; i <= listingEnd * userRatio; i++) {
+    if (callback)  callback(`${chance.first()}\n`);
```
Comment (xueccc) [inline mock-data/generateData.js:74]: Suggestion:  use ternary operator

File: mock-data/generateData.js (hunk 1 above)
Comment (xueccc) [inline mock-data/generateData.js:94]: use ternary operator?

File: database/mongoDb/mongoImportCSV.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  'start_month',
+  'start_date',
+  'duration',
+];
+
+const databaseName = 'booking_service';
+const collectionName = 'bookings';
+const filePath = path.resolve(csvFolderPath, 'testing.csv');
+const commandLine = `mongoimport -d booking_service -c bookings --mode upsert --type csv --file ${filePath} -f ${columns.join(',')}`;
+const mongoImport = async () => {
```
Comment (briansai) [inline database/mongoDb/mongoImportCSV.js:21]: This is awesome.

**Review by briansai — APPROVED** [review]
Looks good. Keep it up!

**Review by xueccc — APPROVED** [review]
Looks good!

File: mock-data/generateData.js (hunk 1 above)
Comment (clfhhc) [reply mock-data/generateData.js:94]: True, I will incorporate that in the next pr

File: mock-data/generateData.js (hunk 3 above)
Comment (clfhhc) [reply mock-data/generateData.js:74]: I will take your advice

**Pull request merged**